engine/src/main/java/io/camunda/zeebe/engine/api/ProcessingScheduleService.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.api;
+
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import java.time.Duration;
+import java.util.function.BiConsumer;
+
+public interface ProcessingScheduleService {
+
+  void runDelayed(Duration delay, Runnable followUpTask);
+
+  <T> void runOnSuccess(ActorFuture<T> precedingTask, BiConsumer<T, Throwable> followUpTask);
```

Comment:
❓ Would it be possible to use `Future` instead of the `ActorFuture` in this method?

Reply:
Not currently. I think after the platform team did its refactoring of actor scheduler it might be possible. Right now we need the `ActorFuture` for the implementation to work. You could use `Future` in the interface and then cast to `ActorFuture` in the implementation, but I felt this is not the right direction to go.